src/frontend/sharedComponents/HookFormTextInput.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import {
+  Control,
+  Controller,
+  FieldValues,
+  Path,
+  useWatch,
+  ValidationRule,
+  useFormState,
+  UseControllerProps,
+} from 'react-hook-form';
+import {TextInput as RNTextInput, StyleSheet, View} from 'react-native';
+import {BLACK, LIGHT_GREY, RED} from '../lib/styles';
+import {ErrorIcon} from './icons';
+import {ViewStyleProp} from '../sharedTypes';
+import {Text} from './Text';
+
+type TextInputProps<InputFields extends FieldValues> = {
+  containerStyle?: ViewStyleProp;
+  showCharacterCount?: boolean;
+  control: Control<InputFields>;
+} & Omit<
+  React.ComponentProps<typeof RNTextInput>,
+  'value' | 'onBlur' | 'onChange'
+> &
+  Omit<UseControllerProps<InputFields>, 'control'>;
+
+type CounterProps<InputFields extends FieldValues> = {
+  name: Path<InputFields>;
+  control: Control<InputFields>;
+  maxLength: ValidationRule<number>;
+  isMaxLengthError: boolean;
+};
+
+/**
+ *
+ * Uses React Hook Form. The name and control should come from the `useForm<T>`. More info here https://www.react-hook-form.com/get-started/#ReactNative. This component also has access to all the same props as React Native's `<TextInput/>`
+ */
+export const HookFormTextInput = <InputFields extends FieldValues>({
+  name,
+  control,
+  rules,
+  containerStyle,
+  showCharacterCount,
+  ...RNInputProp
+}: TextInputProps<InputFields>) => {
+  const error = useFormState({control}).errors[name];
+
+  const maxLengthRule = rules ? rules['maxLength'] : undefined;
+  const maxLength =
+    typeof maxLengthRule === 'number' ? maxLengthRule : maxLengthRule?.value;
+  const errorMessage = error?.message;
+
+  return (
+    <React.Fragment>
+      <View style={[styles.input, containerStyle, error ? styles.error : {}]}>
+        <Controller
+          name={name}
+          control={control}
+          rules={rules}
+          render={({field: {value, onChange, onBlur}}) => (
+            <RNTextInput
+              style={[{flex: 1, color: BLACK}]}
+              value={value}
+              onBlur={onBlur}
+              onChangeText={onChange}
+              {...RNInputProp}
+            />
+          )}
+        />
+        {error && <ErrorIcon style={{position: undefined}} color={RED} />}
+      </View>
+      <View
+        style={[
+          styles.underContainer,
+          {justifyContent: !errorMessage ? 'flex-end' : 'space-between'},
+        ]}>
+        {errorMessage && (
+          <Text style={{color: RED}}>{errorMessage.toString()}</Text>
+        )}
+        {maxLength && showCharacterCount && (
+          <Counter
+            isMaxLengthError={error?.type === 'maxLength'}
+            control={control}
+            name={name}
+            maxLength={maxLength}
+          />
+        )}
+      </View>
```

Comment:
I would consider updating this so that you work with static nodes to reduce some logic:

```tsx
      <View style={[styles.underContainer, {flexDirection: 'row', justifyContent: 'space-between'}]}>
        <View>
          {errorMessage && (
            <Text style={{color: RED}}>{errorMessage.toString()}</Text>
          )}
        </View>

        <View>
          {maxLength && showCharacterCount && (
            <Counter
              isMaxLengthError={error?.type === 'maxLength'}
              control={control}
              name={name}
              maxLength={maxLength}
            />
          )}
        </View>
      </View>
 ```

summary of suggestion:

1. Be explicit about the flex direction (in react native the default is column, which differs from the row default for the web). in this case you want to ensure it's row and not rely on implicit flexbox-spec rules.

2. Set justify content to space-between instead of dynamically determining it. reduces the logic overhead and while not an issue in this case, dynamically calculating layout changes like that is pretty expensive.

3. wrap each node of interest with a view. this ensures that the justify-content from 2 is always being applied on two direct child nodes, regardless of the rendering outcome based on the logic there

Reply:
on additional thought, not as adamant about this suggestion being implemented - just personally find that having to deal with layouts that are conditionally affected in a couple of places can be a little tricky

Reply:
I think your suggestion is a little cleaner. It might cause a slight problem if we were using an inspector like debugger because we would have 2 empty `views`, but I think that would be easy to figure out and none of us use that kind debugger.

Reply:
https://github.com/digidem/CoMapeo-mobile/pull/156/commits/172927279e7822ccb2b3f4496ceafa430ed5b5e6

Reply:
think you still need to add `justifyContent` to the `underContainer` style?

Reply:
https://github.com/digidem/CoMapeo-mobile/pull/156/commits/b4cd6d73ac039f43d2c960a2bbdfadcf3fa420d3